impl/runner.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2025 The Serverless Workflow Specification Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package impl
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/serverlessworkflow/sdk-go/v3/model"
+)
+
+var _ WorkflowRunner = &workflowRunnerImpl{}
+
+type WorkflowRunner interface {
+	GetWorkflowDef() *model.Workflow
+	Run(input interface{}) (output interface{}, err error)
+	GetContext() *WorkflowContext
+}
+
+func NewDefaultRunner(workflow *model.Workflow) WorkflowRunner {
+	wfContext := &WorkflowContext{}
+	wfContext.SetStatus(PendingStatus)
+	// TODO: based on the workflow definition, the context might change.
+	ctx := WithWorkflowContext(context.Background(), wfContext)
+	return &workflowRunnerImpl{
+		Workflow:  workflow,
+		Context:   ctx,
+		RunnerCtx: wfContext,
+	}
+}
+
+type workflowRunnerImpl struct {
+	Workflow  *model.Workflow
+	Context   context.Context
+	RunnerCtx *WorkflowContext
+}
+
+func (wr *workflowRunnerImpl) GetContext() *WorkflowContext {
+	return wr.RunnerCtx
+}
+
+func (wr *workflowRunnerImpl) GetTaskContext() TaskContext {
+	return wr.RunnerCtx
+}
+
+func (wr *workflowRunnerImpl) GetWorkflowDef() *model.Workflow {
+	return wr.Workflow
+}
+
+// Run executes the workflow synchronously.
+func (wr *workflowRunnerImpl) Run(input interface{}) (output interface{}, err error) {
```

Comment:
how this will work?
Ins't a runtime needed?

Reply:
What do you mean? This is the runtime. The `impl`  package implements the spec.

Reply:
curious about the runtime being added in the sdk.